core/src/main/java/org/infinispan/commands/functional/TxReadOnlyKeyCommand.java

```diff
@@ -65,7 +67,7 @@ public Object perform(InvocationContext ctx) throws Throwable {
       if (f != null) {
          ret = f.apply(rw);
       }
-      return snapshot(ret);
+      return snapshot(ret, new CacheEncoders());
```

Comment:
hm, yes and no. I didn't work on the tx part. So encoding + tx are not supported in this PR (no test done on that). Something to be done